src/main/java/uk/gov/hmcts/darts/cases/mapper/CasesMapper.java

```diff
@@ -85,7 +90,11 @@ public ScheduledCase mapToScheduledCase(HearingEntity hearing) {
     public CourtCaseEntity addDefendantProsecutorDefenderJudge(CourtCaseEntity caseEntity, AddCaseRequest caseRequest) {
 
         emptyIfNull(caseRequest.getDefendants()).forEach(newDefendant -> {
-            caseEntity.addDefendant(createNewDefendant(newDefendant, caseEntity));
+            if (newDefendant.matches(unallocatedCaseRegex)) {
```

Comment:
If going with the `Pattern` suggestion then:

```suggestion
            if (unallocatedCaseRegex.matcher(newDefendant).matches()) {
```